examples/api/test/widgets/dismissible/dismissible.1_test.dart

```diff
@@ -0,0 +1,260 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/material.dart';
+import 'package:flutter_api_samples/widgets/dismissible/dismissible.1.dart'
+    as example;
+import 'package:flutter_test/flutter_test.dart';
+
+void main() {
+  Future<void> initWidget(WidgetTester tester) async {
+    await tester.pumpWidget(const example.DismissibleExampleApp());
+  }
+
+  Future<void> reset(WidgetTester tester) async {
+    final Finder fab = find.byType(FloatingActionButton);
+    await tester.tap(fab);
+    await tester.pumpAndSettle();
+  }
+
+  Future<void> dismissHorizontally({
+    required WidgetTester tester,
+    required Finder finder,
+    double dragRatio = 0.8,
+    AxisDirection direction = AxisDirection.left,
+  }) async {
+    final double width = (tester.renderObject(finder) as RenderBox).size.width;
+    final double dx = width * dragRatio;
+
+    final Offset offset = switch (direction) {
+      AxisDirection.left => Offset(-dx, 0.0),
+      AxisDirection.right => Offset(dx, 0.0),
+      _ => throw ArgumentError('$direction is not supported'),
+    };
+
+    await tester.drag(finder, offset);
+  }
+
+  Future<void> testDragBehavior(WidgetTester tester, {
+    required Key key,
+    Matcher expectedBelowThreshold = findsOneWidget,
+    Matcher expectedAboveThreshold = findsNothing,
+  }) async {
+    await tester.scrollUntilVisible(find.byKey(key), 100);
+
+    expect(find.byKey(key), findsOneWidget);
+
+    // Drag below the threshold
+    await dismissHorizontally(
+      tester: tester,
+      finder: find.byKey(key),
+      dragRatio: 0.3
+    );
+
+    await tester.pumpAndSettle();
+
+    expect(find.byKey(key), expectedBelowThreshold);
+
+    await reset(tester);
+
+    // Drag above the threshold
+    await dismissHorizontally(
+      tester: tester,
+      finder: find.byKey(key),
+    );
+
+    await tester.pumpAndSettle();
+
+    expect(find.byKey(key), expectedAboveThreshold);
+  }
+
+  group('Test Drag behaviour', () {
+
+    testWidgets(
+      '0 - Default begavior - `shouldDismiss: null`',
+      (WidgetTester tester) async {
+        await initWidget(tester);
+
+        const ValueKey<int> key = ValueKey<int>(0);
+
+        await testDragBehavior(tester, key: key);
+      },
+    );
```

Comment:
Suggestion for tests in this file, to match spelling/whitespace conventions elsewhere in the framework:

```suggestion
  group('Test Drag behavior', () {
    testWidgets('0 - Default behavior - `shouldDismiss: null`', (WidgetTester tester) async {
      await initWidget(tester);

      const ValueKey<int> key = ValueKey<int>(0);
      await testDragBehavior(tester, key: key);
    });
```